initializr-generator/src/main/resources/templates/Dockerfile

```diff
@@ -0,0 +1,33 @@
+
+# builder image to add AppDynamics agent
+FROM debian as appd
+RUN apt-get update && apt-get install -y --no-install-recommends  unzip \
+        && rm -rf /var/lib/apt/lists/*
+
+# For a more generic build file, use a build ARG for the version of the full file name
+ARG APPD_AGENT_VERSION=4.4.2.22394
+COPY AppServerAgent-${APPD_AGENT_VERSION}.zip /
+RUN unzip -oq AppServerAgent-${APPD_AGENT_VERSION}.zip -d /tmp
+
+# final image
+FROM registry.misys.global.ad/fusion-fabric/centos-fusion:2.1.1-jre
+MAINTAINER olivier.leborne@finastra.com
+EXPOSE 80
+ENV SERVER_PORT 80
+
+COPY --from=appd /tmp /opt/appdynamics
+ENV JAVA_AGENT -javaagent:/opt/appdynamics/javaagent.jar
+
+ENTRYPOINT [ "sh", "-c", \
+    "java -Djava.security.egd=file:/dev/./urandom ${APPDYNAMICS_CONTROLLER_HOST_NAME:+$JAVA_AGENT} ${APPDYNAMICS_NODE_PREFIX:+-Dappdynamics.agent.reuse.nodeName=true -Dappdynamics.agent.reuse.nodeName.prefix=$APPDYNAMICS_NODE_PREFIX} -jar /app.jar" \
+    ]
+# everything above is not impacted by a new build and will be cached by Docker
+
+ARG JAR_FILE
+ADD ${JAR_FILE} app.jar
+
```

Comment:
Please don't forget to use the docker file in the same form as we discussed (commenting out the Dockerfile section for AppDynamics integration).